frontend/src/pages/projects/screens/spawner/storage/usePreferredStorageClass.ts

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+import { AppContext } from '~/app/AppContext';
+import { StorageClassKind } from '~/k8sTypes';
+
+const usePreferredStorageClass = (): StorageClassKind | undefined => {
+  const {
+    dashboardConfig: {
+      spec: { notebookController },
+    },
+    storageClasses,
+  } = React.useContext(AppContext);
+
+  const defaultClusterStorageClasses = storageClasses.filter((storageclass) =>
+    storageclass.metadata.annotations?.['storageclass.kubernetes.io/is-default-class']?.includes(
+      'true',
+    ),
+  );
+
+  const configStorageClassName = notebookController?.storageClassName ?? '';
+
+  if (defaultClusterStorageClasses.length != 0) {
+    return undefined;
+  }
+
+  if (configStorageClassName == '') {
+    return undefined;
+  }
```

Comment:
Oh, technically you can do this at the top -- this is an exit immediately case if things are not configured. Only value doing it late is if you want to see if there is a need for it and log... but you don't seem to be doing that.

Reply:
logic followed was suggested by you, but actually, since later on we check for cases of whether odh config storageclassname is in list of cluster storage classes, returning undefined for this at this position in the flow is correct

```
(using the exit-early strat of functions) Logic can be something like:

    Do we have a default? yes, return undefined
    Do we have a storageClass value in our DashboardConfig? no, return undefined
    Do we have a matching storageClass to the value in the DashboardConfig? no, console.error & return undefined
    Return the Storage Class value you have matching (this allows flexibility of usage, we can determine at this point that we only want the metadata.name if we want)
```

Reply:
Oh check that out -- guess when you're on auto pilot and not writing the code, here we are lol. Nvm! haha

Reply:
There is value of doing this check and returning undefined right after returning undefined if a cluster default (not odh configured) storageclass is set.